browser/brave_search/backup_results_navigation_throttle.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_search/backup_results_navigation_throttle.h"
+
+#include "brave/browser/brave_search/backup_results_service_factory.h"
+#include "brave/components/brave_search/browser/backup_results_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+
+namespace brave_search {
+
+std::unique_ptr<BackupResultsNavigationThrottle>
+BackupResultsNavigationThrottle::MaybeCreateThrottleFor(
+    content::NavigationHandle* navigation_handle) {
+  auto* context = navigation_handle->GetWebContents()->GetBrowserContext();
+  auto* profile = Profile::FromBrowserContext(context);
+  if (!profile->IsOffTheRecord() ||
+      !profile->GetOTRProfileID().IsSearchBackupResults()) {
+    return nullptr;
+  }
+
+  return std::make_unique<BackupResultsNavigationThrottle>(navigation_handle);
```

Comment:
it appears to me that throttles are created for every frame of the doc, but do we need to load subframes at all? Wouldn't they interfere with request mapping logic in the service

Reply:
good catch, let me restrict it to the main frame

Reply:
i assume we need to block loading subframes at all?

Reply:
cancelled all navigation for subframes as discussed in DM